compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1014,6 +1015,21 @@ impl CheckAttrVisitor<'tcx> {
         is_valid
     }
 
+    /// Checks if `#[must_not_suspend]` is applied to a function. Returns `true` if valid.
+    fn check_must_not_suspend(&self, attr: &Attribute, span: &Span, target: Target) -> bool {
+        match target {
+            Target::Fn | Target::Method(..) | Target::ForeignFn | Target::Closure => {
+                self.tcx
+                    .sess
+                    .struct_span_err(attr.span, "`must_not_suspend` attribute should be applied to a struct, enum, `impl Trait`, or `dyn Trait`")
+                        .span_label(*span, "is a function")
```

Comment:
Hmm, shouldn't this check other items besides functions? It seems wrong to apply this to modules/re-exports/constants too.

Reply:
My understanding is that the attribute is already defined in `builtin_attrs.rs` as `Normal`, which already narrows the scope of what it can be applied to? I can add a test and check

Reply:
A test would be great, thanks :)

Reply:
I think this is now covered by the other_items.rs test